store/src/main/java/com/dropbox/android/external/store4/StoreBuilder.kt

```diff
@@ -120,9 +67,69 @@ interface StoreBuilder<Key : Any, Output : Any> {
         @OptIn(ExperimentalTime::class)
         fun <Key : Any, Output : Any> fromNonFlow(
             fetcher: suspend (key: Key) -> Output
-        ): StoreBuilder<Key, Output> = BuilderImpl { key: Key ->
+        ): StoreBuilder<Key, Output> = from(
+            fetcher = fetcher.asFlow()
+        )
+
+        /**
+         * Creates a new [StoreBuilder] from a non-[Flow] fetcher and a [SourceOfTruth].
+         *
+         * Use when creating a [Store] that fetches objects in an HTTP-like single response per
+         * request protocol.
+         *
+         * @param fetcher a function for fetching network records.
+         * @param sourceOfTruth a [SourceOfTruth] for the store.
+         */
+        fun <Key : Any, Input : Any, Output : Any> fromNonFlow(
+            fetcher: suspend (key: Key) -> Input,
+            sourceOfTruth: SourceOfTruth<Key, Input, Output>
+        ): StoreBuilder<Key, Output> = from(
+            fetcher = fetcher.asFlow(),
+            sourceOfTruth = sourceOfTruth
+        )
+
+        /**
+         * Creates a new [StoreBuilder] from a non-[Flow] fetcher.
+         *
+         * Use when creating a [Store] that fetches objects in an HTTP-like single response per
+         * request protocol.
+         *
+         * @param fetcher a function for fetching network records.
+         * @param fetcherTransformer used to translate your fetcher's return value to success value
+         * or error in the case that your fetcher does not communicate errors through exceptions
+         */
+        fun <Key : Any, RawOutput : Any, Output : Any> fromNonFlow(
+            fetcher: suspend (key: Key) -> RawOutput,
+            fetcherTransformer: (RawOutput) -> FetcherResult<Output>
```

Comment:
this is a bit weird., why cant it just provide a FetcherResponse? Name conflict?